cpp/oneapi/dal/algo/kmeans/backend/gpu/kernels_fp_impl.hpp

```diff
@@ -135,8 +142,8 @@ sycl::event kernels_fp<Float>::select(sycl::queue& queue,
 
                 std::int32_t index = -1;
                 Float value = fp_max;
-                for (std::uint32_t i = local_id; i < col_count; i += local_range) {
-                    const Float cur_val = data_ptr[in_offset + i];
+                for (std::uint32_t i = local_id; i < cluster_count; i += local_range) {
```

Comment:
types mismatch: 
i - std::uint32_t i
cluster_count - std::int64_t 
index = i (index is std::int32_t)

Reply:
Fixed